lib/std/os.zig

```diff
@@ -3121,18 +3121,23 @@ pub fn getsockoptError(sockfd: fd_t) ConnectError!void {
     }
 }
 
-pub fn waitpid(pid: i32, flags: u32) u32 {
+
+pub const waitpid_ret = struct {
+   pid: i32,
+   status: u32
+};
+
+pub fn waitpid(pid: i32, flags: u32) waitpid_ret {
     // TODO allow implicit pointer cast from *u32 to *c_uint ?
     const Status = if (builtin.link_libc) c_uint else u32;
     var status: Status = undefined;
     while (true) {
-        switch (errno(system.waitpid(pid, &status, flags))) {
-            0 => return @bitCast(u32, status),
-            EINTR => continue,
-            ECHILD => unreachable, // The process specified does not exist. It would be a race condition to handle this error.
-            EINVAL => unreachable, // The options argument was invalid
-            else => unreachable,
+        var ret: usize = system.waitpid(pid, &status, flags);
+        if (ret == -1) {
```

Comment:
What? Are you comparing unsigned to a negative number? Second, return type of `waitpid` in libc is **signed** (as i said in https://github.com/ziglang/zig/issues/6581#issuecomment-704008582) you **cannot** store it in a `usize` (unsigned)!

Reply:
yeah, sorry that usize thing was fixed and i used an older version of this where it was still not working, i noticed and just pushed the correct version. this indeed was not working and was the code you were actually commenting on in the issue itself. my bad. sorry again. check the latest version instead pls.